doc/example/example_visualization.ipynb

```diff
@@ -2,295 +2,574 @@
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "petab.visualize.plot_problem(petab_problem)\n",
+    "plt.gcf().set_size_inches(10, 4)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Subplots by dataset\n",
     "\n",
+    "We can also plot different datasets (as specified by the optional `datasetId` column in the measurement table) in separate subplots:"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "petab_problem.visualization_df = petab.get_visualization_df(\n",
+    "    example_dir_fujita\n",
+    "    / \"visuSpecs\"\n",
+    "    / \"Fujita_visuSpec_individual_datasets.tsv\"\n",
+    ")\n",
+    "petab_problem.visualization_df"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "petab.visualize.plot_problem(petab_problem)\n",
+    "plt.gcf().set_size_inches(10, 4)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Legend entries\n",
     "\n",
-    "ax = plot_without_vis_spec(\n",
-    "    condition_file_Fujita,\n",
-    "    observable_id_list,\n",
-    "    \"observable\",\n",
-    "    data_file_Fujita,\n",
-    "    plotted_noise=\"provided\",\n",
-    ")"
-   ]
+    "So far, the legend entries don't look very nice. We can change them by specifying the desired labels in the `legendEntries` column:"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "petab_problem.visualization_df = petab.get_visualization_df(\n",
+    "    example_dir_fujita / \"visuSpecs\" / \"Fujita_visuSpec_datasetIds.tsv\"\n",
+    ")\n",
+    "petab_problem.visualization_df"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "petab.visualize.plot_problem(petab_problem)\n",
+    "plt.gcf().set_size_inches(10, 6)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
    "source": [
-    "Plot only simulations"
-   ]
+    "### Plotting individual replicates\n",
+    "\n",
+    "If the measurement file contains replicates, the replicates can also be visualized individually, instead of their means and standard deviations:"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
    "outputs": [],
    "source": [
-    "simu_file_Fujita = \"example_Fujita/Fujita_simulatedData.tsv\"\n",
+    "petab_problem = Problem.from_yaml(petab_yaml_isensee)\n",
+    "petab_problem.visualization_df = petab.get_visualization_df(\n",
```

Comment:
Do you think it is useful to show the visualisation df here as well?

Reply:
Stating what changed should be sufficient.
Not sure if the long tables improve the reading experience.

Out of scope here, but at some point, this should be changed to always just showing 1-2 plots with a clear objective and fewer distractions.

Reply:
In that case we could just remove subplots 3 and 4.

Reply:
Added a comment on what changed. Should be okay for now.